src/widget/wbeatspinbox.cpp

```diff
@@ -21,6 +21,18 @@ WBeatSpinBox::WBeatSpinBox(QWidget* parent,
         : QDoubleSpinBox(parent),
           WBaseWidget(this),
           m_valueControl(configKey, this, ControlFlag::NoAssertIfMissing),
+          beatJumpForward(configKey.group, "beatjump_forward"),
+          beatJumpBackward(configKey.group, "beatjump_backward"),
+          beatJumpSize(configKey.group, "beatjump_size"),
+          beatJumpOneForward(configKey.group, "beatjump_1_forward"),
+          beatJumpOneBackward(configKey.group, "beatjump_1_backward"),
+          beatJumpEightForward(configKey.group, "beatjump_0.125_forward"),
+          beatJumpEightBackward(configKey.group, "beatjump_0.125_backward"),
+          beatLoopEnabled(configKey.group, "loop_enabled"),
+          beatLoopSize(configKey.group, "beatloop_size"),
+          // Hardcoded values for spinbox and loopbox for differentiation between the two
+          sizeSpinBox(60, 26),
+          sizeLoopBox(78, 26),
```

Comment:
The size can not be used as identifier, it's different in each skin that uses this widget. A skin template sets a fixed width, height or both, or a minimum size and some QSizePolicy. Please take a look at the skins, search for `<BeatSpinBox>`.

But you could use `configKey`. That is a `ConfigKey` consisting of a group and a key. Currently used keys are `beatloop_size` and `beatjump_size`.

Reply:
Re Zulip direct message:
I refer to `configkey` in the `WBeatSpinBox` constructor. You can compare the `configkey.key` (QString) with "beatjump_size" and "beatloop_size". Depending on the match you know which spinbox you're deaing with.